### ISSUE TITLE
Lots of issues

### ISSUE THREAD

**Issue by catalin-luntraru** [airbnb/react-dates#140]
Not trying to be negative here, but the documentation is lacking the standard installation procedure.
I had to presume it's

`> npm install --save react-date react-addons-shallow-compare moment`

Any why would I even need `react-addons-shallow-compare`, it's not like I need to run the unit tests on this library myself.

Second, there's no documentation to set the code in your app. I had to figure it out myself that you have to write

```
import { SingleDatePicker } from 'react-dates';
import 'react-dates/css/variables.scss';
import 'react-dates/css/styles.scss';

<SingleDatePicker {...this.props} id="date_input" date={null} focused={false} onDateChange={this.onDateChange} onFocusChange={this.onFocusChange} />
```

And third, the div layer with the calendar is hidden so I had to disable `visibility: hidden` with chrome dev tools to at least see what this calendar is all about. Every time I would hover over a calendar date it would take 40-50% of my quad core CPU and I'm running Lubuntu, Probably on windows, the CPU would run out the window.

I'm amazed why this library has 2.5k stars, unless it's just that people know about airbnb.

TLDR: This library needs basic documentation for install and usage, not an api dump. And performance needs a serious check.

**Comment by ljharb** [comment]
Regarding the first item, the documentation tells you exactly how to install - run the command [here](https://github.com/airbnb/react-dates#how-to-use).

**Comment by catalin-luntraru** [comment]
I said it lacks a standard installation procedure.
I don't think this `npm install --save react-dates moment@>=#.## react@>=#.## react-dom@>=#.## react-addons-shallow-compare@>=#.##` is standard.
Or this:

```
 (
    export PKG=react-dates;
    npm info "$PKG" peerDependencies --json | command sed 's/[\{\},]//g ; s/: /@/g; s/ *//g' | xargs npm install --save "$PKG"
  )
```

**Comment by ljharb** [comment]
@catalin-luntraru that latter script IS the standard installation procedure, because there does not exist any other way of automatically installing the proper peer dependencies into your package.json. The former, as it says, is an example of the command that the latter generates.

A naive `npm install --save` command for a module that has peerDependencies is highly likely to install the wrong versions, which is why that shell script exists.

**Comment by ljharb** [comment]
As for `react-addons-shallow-compare`, that is required, at runtime, by React itself, so yes, you do need it installed.

**Comment by majapw** [comment]
The examples folder of the repo has pretty basic components (`SingleDatePickerWrapper` and `DateRangePickerWrapper`) you can use to help construct a datepicker in your existing framework and the storybook itself has tens of examples of different configurations you can use. We are slowly adding in more instructions to the README, but as always, PRs are definitely welcome (especially for setups that differ from ours!).

As for the hidden div, it seems like you mean the calendar dropdown. If you hook up your `onFocusChange` listener as shown in the example here (https://github.com/airbnb/react-dates/blob/master/examples/SingleDatePickerWrapper.jsx), the div will show when you focus on the input. This should also be the behavior in the public storybook at http://www.airbnb.io/react-dates

Performance is something that we're actively working on. I think especially on phones and older computers, the work that's being done to style the calendar days themselves is excessive and we've been looking at pulling the modifiers logic out of `CalendarMonth` and into something like a `CalendarDayWrapper` component. We've noticed when looking at the perf that this is a source of a lot of extraneous render calls. I think this will help a lot in this department, although I can't say I've ever seen the hover logic take up 50% of someone's CPU. Can you provide any more specs on your system so we can get a sense of what is going on?

**Comment by majapw** [comment]
I also have to say that this issue description seems unnecessarily negative. We're constantly trying to improve the repository and are definitely open to constructive criticism, but leading with "Worst library so far" is unlikely to get you the support you want. No one is forcing you to use this library over any other datepickers out there; we're just trying to demo what we have and collaborate with others to improve on it. ✌️

**Comment by catalin-luntraru** [comment]
To check for the 50% CPU issue, you would have to install and use it as I stated in my previous post. If you use the exact same code and then disable the visibility:hidden option and then start hovering on the dates, at least my i3 CPU 2.9Ghz on Lubuntu runs at 40-50% usage.

About negativity, indeed I should have toned it down a bunch. I did however edited the title, but i guess the notification emails were already sent

**Comment by catalin-luntraru** [comment]
@ljharb  Why would you even need dependencies for an exact version of a package? This opens up the can of worms that ruby has, and sooner or later we would also have to install a version manage on nodejs to support different versions for different projects.

**Comment by ljharb** [comment]
@catalin-luntraru the way npm works is that singletons - like react and *literally everything in the react ecosystem* - require use of peerDependencies. Whether you like it or not, *this is how both npm and React work* and if you don't like it, tough.

**Comment by ljharb** [comment]
@catalin-luntraru i'm going to go a little farther here, and warn you that if you continue your sanctimonious unwarranted negativity, whilst not even understanding the very ecosystems in which this project operates, user moderation will be in order.

Thanks for your report - I don't think we need any further information to look into this, so I'm going to lock this particular conversation from further comment.

**Comment by majapw** [comment]
Hur dur, v11.0.0 (I swear the last breaking release for a while) rearchitects modifiers so now there's about the same overhead at mount, but then calendar interactions only updates the relevant `CalendarDay` objects instead of all of them.

You can see more details in #450, but in the meantime, I'm going to close this issue of being "generally slow". There are still some improvements to be made, notably during the month transition phase, but I think this new architecture opens up a lot of doors. :)
